src/App.tsx

```diff
@@ -1,24 +1,211 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
+import {
+  getTodos,
+  deleteTodo,
+  postTodo,
+  updateTodo,
+} from './api/todos';
+import { ErrorMessage } from './components/ErrorMessage/ErrorMessage';
+import { TodoFilter } from './components/TodoFilter/TodoFilter';
+import { Header } from './components/Header/Header';
+import { Loader } from './components/Loader/Loader';
+import { TodoList } from './components/TodoList/TodoList';
+import { USER_ID } from './types';
+import { TodosError } from './types/Errors';
+import { Todo } from './types/Todo';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { filterTodos } from './helpers';
+import { SortTodoBy } from './types/SortTodoBy';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const [sortTodosBy, setSortTodosBy] = useState(SortTodoBy.Default);
+  const [loadingTodosIds, setLoadingTodosIds] = useState<number[]>([]);
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [loadingError, setLoadingError] = useState(false);
+  const [errorText, setErrorText] = useState('');
+
+  const [disableField, setDisableField] = useState(false);
+
+  const visiableTodos = filterTodos(todos, sortTodosBy);
+  const completedTodos = useCallback(() => {
+    return todos.filter(todo => todo.completed);
+  }, [todos]);
+  const activeTodos = todos.filter(todo => !todo.completed);
+  const hasActiveTodos = activeTodos.length !== 0;
+  const hasCompletedTodos = completedTodos().length !== 0;
+
+  const handleShowError = (text: string) => {
+    setLoadingError(true);
+    setErrorText(text);
```

Comment:
you don't need boolean state for error, you already have state with message - just create TodosError.None and set it by default